hpat/tests/test_series.py

```diff
@@ -417,76 +417,106 @@ def test_impl(A):
+        for operator in arithmetic_binops:
+            func_text = "def test_impl(S, i):\n"
+            func_text += "   return S {} i\n".format(operator)
+            loc_vars = {}
+            exec(func_text, {}, loc_vars)
+            test_impl = loc_vars['test_impl']
+            hpat_func = hpat.jit(test_impl)
+        
+            n = 11
+            df = pd.DataFrame({'A': np.arange(1, n)})
+            pd.testing.assert_series_equal(hpat_func(df.A, 1), test_impl(df.A, 1), check_names=False)
 
     def test_series_op3(self):
-        def test_impl(A, i):
-            A += i
-            return A
-        hpat_func = hpat.jit(test_impl)
+        arithmetic_binops = ('+', '-', '*', '/', '//', '%', '**')
 
-        n = 11
-        df = pd.DataFrame({'A': np.arange(n)})
-        pd.testing.assert_series_equal(hpat_func(df.A.copy(), 1),
-                                       test_impl(df.A, 1), check_names=False)
+        for operator in arithmetic_binops:
+            func_text = "def test_impl(S, i):\n"
+            func_text += "   S {} i\n".format(operator)
+            func_text += "   return S\n"
+            loc_vars = {}
+            exec(func_text, {}, loc_vars)
+            test_impl = loc_vars['test_impl']
+            hpat_func = hpat.jit(test_impl)
+        
+            n = 11
+            df = pd.DataFrame({'A': np.arange(1, n)})
+            pd.testing.assert_series_equal(hpat_func(df.A, 1), test_impl(df.A, 1), check_names=False)
 
     def test_series_op4(self):
-        def test_impl(A):
-            return A.add(A)
-        hpat_func = hpat.jit(test_impl)
+        arithmetic_methods = ('add', 'sub', 'mul', 'div', 'truediv', 'floordiv', 'mod', 'pow')
 
-        n = 11
-        A = pd.Series(np.arange(n))
-        pd.testing.assert_series_equal(hpat_func(A), test_impl(A))
+        for method in arithmetic_methods:
+            func_text = "def test_impl(S):\n"
+            func_text += "   return S.{}(S)\n".format(method)
+            loc_vars = {}
+            exec(func_text, {}, loc_vars)
+            test_impl = loc_vars['test_impl']
+            hpat_func = hpat.jit(test_impl)
+        
+            n = 11
+            df = pd.DataFrame({'A': np.arange(1, n)})
+            pd.testing.assert_series_equal(hpat_func(df.A), test_impl(df.A), check_names=False)
 
     def test_series_op5(self):
         def test_impl(A):
-            return A.pow(A)
+            return -A
         hpat_func = hpat.jit(test_impl)
 
         n = 11
         A = pd.Series(np.arange(n))
         pd.testing.assert_series_equal(hpat_func(A), test_impl(A))
 
     def test_series_op6(self):
-        def test_impl(A, B):
-            return A.eq(B)
-        hpat_func = hpat.jit(test_impl)
-
-        n = 11
-        A = pd.Series(np.arange(n))
-        B = pd.Series(np.arange(n)**2)
-        pd.testing.assert_series_equal(hpat_func(A, B), test_impl(A, B))
+        comparison_binops = ('<', '>', '<=', '>=', '!=', '==')
```

Comment:
Do we want to check the case `if a=b` and evaluate properly throw exception?

Reply:
Hmm, I didn't think of it. But it as far as I understand it won't test HPAT anyway, it will test Python, because SyntaxError will be raised before we attempt to jit-compile anything. Am I wrong?

Reply:
I don't know. I just propose the idea to implement additional checks. It is ok if you think it is not applicable here.